pkg/tenancy/tenancy_test.go

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package tenancy_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+
+	"github.com/efficientgo/core/testutil"
+	"github.com/thanos-io/thanos/pkg/component"
+	"github.com/thanos-io/thanos/pkg/store"
+	"github.com/thanos-io/thanos/pkg/store/storepb"
+	"github.com/thanos-io/thanos/pkg/tenancy"
+
+	"github.com/pkg/errors"
+
+	storetestutil "github.com/thanos-io/thanos/pkg/store/storepb/testutil"
+)
+
+// mockedStoreAPI is test gRPC store API client.
+type mockedStoreAPI struct {
+	RespSeries      []*storepb.SeriesResponse
+	RespLabelValues *storepb.LabelValuesResponse
+	RespLabelNames  *storepb.LabelNamesResponse
+	RespError       error
+	RespDuration    time.Duration
+	// Index of series in store to slow response.
+	SlowSeriesIndex int
+
+	LastSeriesReq      *storepb.SeriesRequest
+	LastLabelValuesReq *storepb.LabelValuesRequest
+	LastLabelNamesReq  *storepb.LabelNamesRequest
+
+	t *testing.T
+}
+
+const testTenant = "test-tenant"
+
+func (s *mockedStoreAPI) Info(context.Context, *storepb.InfoRequest, ...grpc.CallOption) (*storepb.InfoResponse, error) {
+	return nil, status.Error(codes.Unimplemented, "not implemented")
+}
+
+func (s *mockedStoreAPI) Series(ctx context.Context, req *storepb.SeriesRequest, _ ...grpc.CallOption) (storepb.Store_SeriesClient, error) {
+	return nil, status.Error(codes.Unimplemented, "not implemented")
+}
+
+func (s *mockedStoreAPI) LabelNames(_ context.Context, req *storepb.LabelNamesRequest, _ ...grpc.CallOption) (*storepb.LabelNamesResponse, error) {
+	return nil, status.Error(codes.Unimplemented, "not implemented")
+}
+
+func (s *mockedStoreAPI) LabelValues(ctx context.Context, req *storepb.LabelValuesRequest, _ ...grpc.CallOption) (*storepb.LabelValuesResponse, error) {
+	fmt.Println("Mocked label values call")
```

Comment:
Let's not add `fmt.Print*`. I'm surprised the linter hasn't caught this :thinking: